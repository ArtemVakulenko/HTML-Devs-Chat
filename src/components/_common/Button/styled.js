import styled from 'styled-components';

export const StButton = styled.button`
    color:  ${({ colors, theme }) => colors[theme].buttonText};
    height: 2rem;
    font-size: 1rem;
    margin: 1rem;
    padding: 0.25rem 1rem;
    border: 2px solid  ${({ colors, theme }) => colors[theme].button};
    border-radius: 10px;
    background-color: ${({ colors, theme }) => colors[theme].button};
    cursor: pointer;
    :hover {
        border: 2px solid  ${({ colors, theme }) => colors[theme].main};
    }
`;
export const StIcon = styled.button`
    height: 40px;
    width: 40px;
    margin: 1rem;
    padding: 0.25rem 1rem;
    border: none;
    border-radius: 10px;
    background: url(${(props) => props.icon}) no-repeat;
    cursor: pointer;
    background-size: contain;
    :hover {
        border: 2px solid  transparent
    }
`;
