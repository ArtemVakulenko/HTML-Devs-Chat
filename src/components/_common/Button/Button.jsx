import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StButton, StIcon } from './styled';
import { useTheme } from '../../hooks';

const Button = ({ value, icon, onClick }) => {
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  return icon ? (
    <StIcon
      colors={colors}
      theme={theme}
      icon={icon}
      onClick={onClick}
    />
  ) : (
    <StButton
      colors={colors}
      theme={theme}
      onClick={onClick}
    >
      {t(value)}
    </StButton>
  );
};

Button.propTypes = {
  icon: PropTypes.string,
  value: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
