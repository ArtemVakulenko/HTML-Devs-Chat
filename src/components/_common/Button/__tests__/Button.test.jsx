import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
      t: jest.fn(),
    })),
  }));
  jest.mock('../../../hooks', () => ({
    useTheme: jest.fn().mockImplementation(() => ({
      colors: {
        dark: {
          color: '',
          input: '',
          placeholder: '',
        },
      },
      theme: 'dark',
    })),
  }));

describe('Button', () => {
      const props = {
        value: 'login',
        onClick: jest.fn(),
      };
      const props2 = {
        icon: 'login',
        onClick: jest.fn(),
      };
    it('Should match snapshot', () => {
        const component = shallow(<Button {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('Should have button', () => {
        const component = global.mount(<Button {...props} />);
        expect(component.find('button')).toHaveLength(1);
    });
    
    it('Should have Icon', () => {
        const component = global.mount(<Button {...props2} />);
        expect(component.find('button')).toHaveLength(1);
    });
    it('Should be onclick', () => {
        const component = global.mount(<Button {...props} />);
        component.find('button').simulate('click');
        expect(component.props().onClick).toHaveBeenCalled();
      });
});
