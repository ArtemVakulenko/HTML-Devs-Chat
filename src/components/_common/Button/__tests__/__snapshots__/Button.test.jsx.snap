// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<button class=\\"styled__StButton-iwFefC fohYxd\\"></button>"`;
